### PR TITLE
added 2 overrided methods to Rectangle class

### PR DESCRIPTION
**Pull request by MarberyUA**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,20 @@ public Integer getLength() {
...
+    public boolean equals(Object obj) {
+        return obj instanceof Rectangle && (this.getColor().equals(((Rectangle) obj).getColor())
+                && this.getLength().equals(((Rectangle) obj).getLength())
+                && this.getWidth().equals(((Rectangle) obj).getWidth()));
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 3;
+        result = result * 11 + width.hashCode();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: what happens, if one of the fields == null ?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,20 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        return obj instanceof Rectangle && (this.getColor().equals(((Rectangle) obj).getColor())
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: should we use it instanceof or getClass() and why? 
+ there should be 4 checks in total, you have 2 now

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,39 @@ public Integer getLength() {
...
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Rectangle)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: should we use here getClass() or instanceof ? 
and why

Comment (MarberyUA) [reply src/main/java/core/basesyntax/Rectangle.java]: We should use getClass() because we can get troubles when this class get child class

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**